scripts/archive_collection.py

```diff
@@ -17,83 +17,76 @@
 $ python scripts/archive_collection.py \
     data/proceedings.json \
     data/conferences.json \
-    dev \
+    --output_file updated-proceedings.json \
+    --stage dev \
     --verbose 50 \
     --num_cpus -2 \
     --max_items 10
 ```
 """
 import argparse
-import io
 from joblib import Parallel, delayed
 import json
 import logging
 import os
 import random
-import requests
 import sys
 
-import zen
+import zen.api
+import zen.models
 
-logger = logging.getLogger("fetch")
+logger = logging.getLogger("archive-collection")
 
-META = {
-    "license": "CC-BY-4.0",
-    "access_right": "open",
-    "description": "<p></p>",
-    "communities": [{"identifier": "ismir"}],
-    "imprint_publisher": "ISMIR",
-    "upload_type": "publication",
-    "publication_type": "conferencepaper"
-}
 
-
-def build_record(record, conferences):
-    record = dict(**record)
-    key = record['conference_acronym']
-    meta = META.copy()
-    meta.update(**conferences[key])
-    meta["partof_title"] = meta.pop('conference_title')
-    meta['conference_title'] = meta["partof_title"].split("the ")[-1]
-    meta["imprint_place"] = meta["conference_place"]
-    res = requests.get(record.pop('pdf_url'))
-    fp = io.BytesIO(res.content)
-    meta.update(**record)
-    return os.path.basename(res.url), fp, meta
-
-
-def upload(record, conferences, stage):
+def upload(ismir_paper, conferences, stage=zen.DEV):
     """Upload a file / metadata pair to a Zenodo stage.
 
     Parameters
     ----------
-    filename : str
-        Path to a local file on disk.
-        TODO: Could be a generic URI, to allow webscraping at the same time.
+    ismir_paper : zen.models.IsmirPaper
+        ISMIR paper record.
 
-    metadata : dict
-        Metadata associated with the resource.
+    conferences : dict of zen.models.IsmirConference
+        Conference metadata.
 
     stage : str
         One of [dev, prod]; defines the deployment area to use.
 
-    zid : str, default=None
-        If provided, attempts to update the resource for the given Zenodo ID.
+    Returns
+    -------
+    updated_paper : zen.models.IsmirPaper
+        An updated IMSIR paper object.
     """
-    if not record['pdf_url'].lower().endswith('.pdf'):
-        return
+    ismir_paper = zen.models.IsmirPaper(**ismir_paper)
+    conf = zen.models.IsmirConference(**conferences[ismir_paper['year']])
+
+    if not ismir_paper['zenodo_id']:
+        # New submission
+        zid = zen.create_id(stage=stage)
+    else:
+        # Update mode
+        #  * If the checksum is different, re-upload the pdf
```

Comment:
actually, Zenodo will do this anyways, at the cost of transferring the 4MB or whatever

Reply:
Per paper, right?
And it requires that you have a PDF of the paper. Would be cool to change metadata without downloading all proceedings first.

Reply:
yea, will add an issue for that ... there are a few ways we could tackle that, though it's an optimization and unclear how often this will happen